src/Metalama.Patterns.Contracts/InvariantSuspensionCounter.cs

```diff
@@ -1,20 +1,36 @@
 ﻿// Copyright (c) SharpCrafters s.r.o. See the LICENSE.md file in the root directory of this repository root for details.
 
+using JetBrains.Annotations;
+
 namespace Metalama.Patterns.Contracts;
 
+/// <summary>
+/// A class used by the <see cref="InvariantAttribute"/> aspect to count the number of times the <c>SuspendInvariants</c>
+/// method has been invoked.
+/// </summary>
+[PublicAPI]
 public sealed class InvariantSuspensionCounter : IDisposable
 {
     private int _value;
 
+    /// <summary>
+    /// Decrements the counter. Note that this does not verify invariants when the counter is back to zero.
+    /// </summary>
     public void Dispose()
     {
         Interlocked.Decrement( ref this._value );
     }
```

Comment:
Note that this implementation violates the `IDisposable` contract that:

> If an object's `Dispose` method is called more than once, the object must ignore all calls after the first one.

Though fixing that would probably require an allocation and I don't know if that's worth it.

Reply:
Yeah saving allocation was my design goal. I have changed that to a disposable struct.